CMakeLists.txt

```diff
@@ -14,14 +14,19 @@ PROJECT(Descent3)
 IF (UNIX)
 SET (D3_GAMEDIR "~/Descent3/")
 
+  IF (APPLE)
+    SET(BITS "-m64")
+    SET(EXTRA_CXX_FLAGS "-Wno-address-of-temporary")
+  ELSE()
+    SET(BITS "-m32")
+    SET(EXTRA_CXX_FLAGS "-fpermissive")
+  ENDIF()
+
   SET(CMAKE_CXX_COMPILER "g++")
-  SET(CMAKE_CXX_FLAGS "-O0 -g -Wno-write-strings -Wno-multichar -Wno-address-of-temporary")
-  SET(CMAKE_C_FLAGS "-O0 -g")
+  SET(CMAKE_CXX_FLAGS "-O0 -g -std=c++14 -Wno-write-strings -Wno-multichar ${BITS} ${EXTRA_CXX_FLAGS}")

```

Comment:
You need this done like this
```
set(CMAKE_CXX_STANDARD 14)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_EXTENSIONS OFF)
```
But there some issues in with CMake compatibility level. Also, as agreed in #4, there should be C++17 standard support.

Reply:
Done, plus a few lines of code adapted. And indeed, that is just functional cmake to get the code to compile.